### ISSUE TITLE
No Crimson Hearts in Corruption world with Crimson chasm

### ISSUE THREAD

**Issue by nlotra** [JavidPack/TerraCustom#8]
Cross-posting the issue from the terraria forums for visibility. 

I've generated upwards of 25 worlds with the "Corruption plus Crimson chasm setting" and every one has only shadow orbs. I'm fairly certain this setting is bugged to only generate shadow orbs and no crimson hearts. 

I've had some mixed results when generating "Crimson plus Corruption chasm" worlds where it will range from having only hearts to 1:3 ratio of orbs to hearts. It seems to heavily favour generating the type that matches the world type. Can something be done to make both types always generate both types of orbs/hearts, or even the option to favour one over the other? I don't care much if they generate in their respective biome or are mixed together but I feel this is the main issue with creating worlds with both. 

Another possibly related issue is on any world with both crimson and corruption, corruption seems to be generated after the crimson regardless which of the options for both are selected. Oftentimes the corruption overlaps the crimson and so the crimson chasm gets blocked. I think due to the design of the crimson chasm it should be generated after the corruption because it only has one entrance on the surface and can overlap the corruption without as much of an issue. Of course this could also be an option if people like the way it's currently generated.

**Comment by nlotra** [comment]
I wanted to have a go at patching this but seems it doesn't like the latest version of Terraria.

**Comment by JavidPack** [comment]
Yeah, 1.3.5.3 is a huge mess to decompile until ilspy is updated, so you need the 1.3.5.2 exe for the patcher.

Easiest way to get it is to follow this guide: https://www.reddit.com/r/Steam/comments/611h5e/guide_how_to_download_older_versions_of_a_game_on/

**Comment by nlotra** [comment]
Thanks for that, I've been messing with the code for a bit and as I suspected, if you move the corruption generation to be before the crimson, you get the hearts and orbs in the right place. I also had removed the same line in the cleanup as I've just noticed you've done in your most recent commit. I need some time to do some further testing with this but I'll probably send a pull request your way if there's no further issues.
